up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto

```diff
@@ -14,340 +14,129 @@
 syntax = "proto3";
 
 package uprotocol.core.udiscovery.v3;
-import "google/protobuf/timestamp.proto";
 import "uprotocol/uoptions.proto";
 import "uprotocol/v1/uri.proto";
 
 option java_package = "org.eclipse.uprotocol.core.udiscovery.v3";
 option java_outer_classname = "UDiscoveryProto";
 option java_multiple_files = true;
 
-// Platform USE Discovery Service Interface
+// Client facing APIs to the uProtocol Discovery Service.
+// 
+// The uDiscovery service is used to store information about devices and services namely their
+// addresses (in URI format), and properties of topics. This API is used by clients (uEs) to be
+// about to access the information within the database. Clients talk to their local uDiscovery service
+// who, when is unable to find what it is looking for, queries the central database to find the information.
+// UDiscovery works in heiarchial mode where the local uDiscovery service is the first point of contact
+// and the domain level service (for information about devices within a domain), and then central service
+// is the last point of contact.
+//
+// NOTE: the internal uDiscovery communication protocol (for how data is replicated between the local, domain,
+// and central) is NOT covered in this interface as this is ONLY the client APIs.
 service uDiscovery {
   option (uprotocol.service_name) = "core.udiscovery"; // Service name
   option (uprotocol.service_version_major) = 3;
   option (uprotocol.service_version_minor) = 0;
   option (uprotocol.service_id) = 1;
 
-  // uDiscovery Node Change Notification that sends the Update message
-  option (uprotocol.notification_topic) = {
-    id: 0x8000,
-      name: "NodeChange",
-      message: "Notification"
-  };
 
-    
-
-  // Used by any uProtocol application or service to find service instances
-  // locations. The passed UUri contains valid UEntity, UResource, and UAuthority information
-  // for a query.
-  // What is returned is a list of UUris that match the query.
-  // If lookup fails, the response will be a UStatus with
-  // 2. code.NOT_FOUND: No matching UUris were found
-  // 3. code.INVALID_ARGUMENT: The passed UUri is invalid
-  // 4. code.PERMISSION_DENIED: The caller does not have permission to perform the query
-  rpc LookupUri(LookupUriRequest) returns (LookupUriResponse) {
+  // Find Services. 
```

Comment:
FMPOV all of this documentation should go into the specification document ...

Reply:
@sophokles73 I changed this now so please re-review